index.js

```diff
@@ -134,11 +134,40 @@ var Router = function(){
 
       var route = Route(path);
       route.fn = fn;
+      route.idx = this.routes.length;
 
       this.routes.push(route);
       this.routeMap[path] = fn;
     },
 
+    removeRoute: function(path) {
+      if (!path) throw new Error(' route requires a path');
+      if (!this.routeMap[path]) {
+        throw new Error('path does not exist: ' + path);
+      }
+
+      // find route for this specific path, not necessarily the first that
+      // matches the pattern
+      var match;
+      this.routes.forEach(function eachRoute(route) {
+        if (route.src === path) {
+          match = route;
+        }
+      });
+
+      if (match) {
+        this.routes.splice(match.idx, 1);
```

Comment:
`match.idx` is set inside the loop in `addRoute` and it's accurate. Not sure I follow your point here. Also, I can't use `indexOf` here because I just have `path` which is a string and not a valid route object.